pattoo/api/portal/purge.py

```diff
@@ -0,0 +1,102 @@
+"""Routes for purge."""
+
+# Flask imports
+from flask import Blueprint
+from flask import request, session
+
+# Standard import
+from sqlalchemy import func
+import datetime
+import time
+
+# Pattoo imports
+from pattoo.db import db
+from pattoo.db.models import Data as DataModel
+
+# Import server resources
+from .forms import PurgeForm
+
+# Define the AGENTS global variable
+PURGE = Blueprint('PURGE', __name__)
+
+@PURGE.route('/api/purge', methods=['GET', 'POST'])
+def datapoints():
+    """
+    Handle purging of data.
+
+    Args:
+        None
+
+    Returns:
+        response (dict): Response Message
+
+    """
+    # Check if a user was stored in session first
+    if session.get('idx_user', None) is None:
+        response = {'data': {'message': 'Login first'}}
+        # Block access if no user was found in session
+        return response
+
+    response = {'data':{'message': 'Query did not run'}}
+    with db.db_query(20192, close=False) as db_session:
+        # Get the minimum timestamp of Data, which is the oldest timestamp
+        oldest_data = db_session.query(func.min(DataModel.timestamp)).one()
+        # oldest_data => (timestamp,)
+        # oldest_data[0] => timestamp
+        oldest_day = oldest_data[0]
+
+        # Get timestamp of the moment and translate to the timestamp
+        # form in the Data model
+        _today = time.time() * 1000
+        # Get different of the moment's timestamp and the oldest timestamp
+        # from the Data model
+        _difference = _today - oldest_day
+
+        # Convert the difference into normal timestamp form then
+        # convert into datetime object to retrieve the difference in the
+        # number of days
+        oldestDay = (datetime.datetime.fromtimestamp((_difference / 1000))).day
+
+        response = {'data':{'oldestDay': oldestDay, 'message': 'Query ran'}}
```

Comment:
Return `response` immediately

Reply:
If I do that, it would never be able to process a POST request. `oldestDay` is a variable I would always want to have access to. So that's why I didn't put the above statements in the GET control structure.